src/Umbraco.Core/Configuration/HealthChecks/CompilationDebugCheck.cs

```diff
@@ -1,33 +1,39 @@
 ﻿using System.Collections.Generic;
-using Umbraco.Core.Hosting;
-using Umbraco.Core.IO;
+using Microsoft.Extensions.Options;
+using Umbraco.Core.Configuration.Models;
 using Umbraco.Core.Logging;
 using Umbraco.Core.Services;
 
-namespace Umbraco.Web.HealthCheck.Checks.Config
+namespace Umbraco.Core.HealthCheck.Checks.LiveEnvironment
 {
     [HealthCheck("61214FF3-FC57-4B31-B5CF-1D095C977D6D", "Debug Compilation Mode",
         Description = "Leaving debug compilation mode enabled can severely slow down a website and take up more memory on the server.",
         Group = "Live Environment")]
-    public class CompilationDebugCheck : AbstractConfigCheck
+    public class CompilationDebugCheck : AbstractSettingsCheck
     {
-        public CompilationDebugCheck(ILocalizedTextService textService, IHostingEnvironment hostingEnvironment, ILogger logger)
-            : base(textService, hostingEnvironment, logger)
-        { }
+        private readonly HostingSettings _hostingSettings;
 
-        public override string FilePath => "~/Web.config";
+        public CompilationDebugCheck(ILocalizedTextService textService, ILogger logger,IConfigurationService configurationService, IOptions<HostingSettings> hostingSettings)
+            : base(textService, logger, configurationService)
+        {
+            _hostingSettings = hostingSettings.Value;
+        }
```

Comment:
Here do we need to inject IOptionsMonitor<HostingSettings> to ensure the value is updated on the fly

Also save the  IOptionsMonitor<HostingSettings> in the field and below use `_hostingSettings.CurrentValue.Debug` instead

Reply:
Will do!